File: SQL/Select/SQLCondition.vb

' ___________________________________________________
'
'  © Hi-Integrity Systems 2010. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On 

Namespace SQL

    Public Class SQLCondition

        Private pstrFieldName As String = String.Empty
        Private pobjValue As Object
        Private peCompare As ComparisonOperator
        Private pobjTable As SQLSelectTable

        Friend Sub New()

        End Sub

        Public Sub New( _
            ByVal strFieldName As String, _
            ByVal eCompare As ComparisonOperator, _
            ByVal objValue As Object)

            Me.FieldName = strFieldName
            Me.Compare = eCompare
            pobjValue = objValue

        End Sub

        Public Property Table() As SQLSelectTable
            Get

                Return pobjTable

            End Get

            Set(ByVal Value As SQLSelectTable)

                pobjTable = Value

            End Set
        End Property

        Public Property FieldName() As String
            Get

                Return pstrFieldName

            End Get

            Set(ByVal Value As String)

                pstrFieldName = Value

            End Set
        End Property

        Public Property Compare() As ComparisonOperator
            Get

                Return peCompare

            End Get

            Set(ByVal Value As ComparisonOperator)

                peCompare = Value

            End Set
        End Property

        Public Property Value() As Object
            Get

                Value = pobjValue

            End Get

            Set(ByVal Value As Object)

                pobjValue = SQLConditionValue(Value)

            End Set
        End Property

        Friend ReadOnly Property SQL(ByVal eConnectionType As Database.ConnectionType) As String
            Get

                CompareValuePairAssertValid(Me.Compare, pobjValue)

                Return _
                    SQLFieldNameAndTablePrefix(Me.Table, Me.FieldName, eConnectionType) & " " & _
                    SQLConvertCondition(Me.Compare, Me.Value, eConnectionType)

            End Get
        End Property

    End Class

End Namespace

File: SQL/Select/SQLCondition.cs
// ___________________________________________________
//
//  © Hi-Integrity Systems 2010. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// ___________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLCondition
	{
		private string pstrFieldName = string.Empty;
		private object pobjValue;
		private ComparisonOperator peCompare;
		private SQLSelectTable pobjTable;
			
		internal SQLCondition()
		{
		}
			
		public SQLCondition(string strFieldName, ComparisonOperator eCompare, object objValue)
		{
			this.FieldName = strFieldName;
			this.Compare = eCompare;
			pobjValue = objValue;
		}
			
		public SQLSelectTable Table
		{
			get
			{
				return pobjTable;
			}
				
			set
			{
				pobjTable = value;
			}
		}
			
		public string FieldName
		{
			get
			{
				return pstrFieldName;
			}
				
			set
			{
				pstrFieldName = value;
			}
		}
			
		public ComparisonOperator Compare
		{
			get
			{
				return peCompare;
			}
				
			set
			{
				peCompare = value;
			}
		}
			
		public object Value
		{
			get
			{
				return pobjValue;
			}
				
			set
			{
				pobjValue = GetConditionValue(value);
			}
		}

		internal static object GetConditionValue(object objValue)
		{
			if (objValue is SQLFieldValue)
				return ((SQLFieldValue)objValue).Value;
			else
				return objValue;
		}
	}
}
